File: emegencyQueue/src/main/java/com/douglas/unisales/emegencyQueue/model/domain/MedicalCare.java
package com.douglas.unisales.emegencyQueue.model.domain;

import java.util.UUID;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "medical_care")
public class MedicalCare {

    @Id
    private UUID id = UUID.randomUUID();

    private String floor;
    private String roomNumber;
    private Boolean isICU;

    @ManyToOne
    @JoinColumn(name = "patient_id")
    private Patient patient;

    @ManyToOne
    @JoinColumn(name = "status_id")
    private Status status;

    @ManyToOne
    @JoinColumn(name = "service_phase_id")
    private ServicePhase servicePhase;
}

File: HospitalApiGo/HospitalSantaMônica/go-api/internal/models/medical_care.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type MedicalCare struct {
	ID            uuid.UUID    `json:"id" gorm:"type:uuid;primary_key;default:uuid_generate_v4()"`
	Floor         string       `json:"floor" gorm:"not null"`
	RoomNumber    string       `json:"roomNumber" gorm:"not null"`
	IsICU         bool         `json:"isICU" gorm:"not null"`
	PatientID     uuid.UUID    `json:"patientId" gorm:"type:uuid;not null"`
	Patient       Patient      `json:"patient" gorm:"foreignKey:PatientID"`
	StatusID      uuid.UUID    `json:"statusId" gorm:"type:uuid;not null"`
	Status        Status       `json:"status" gorm:"foreignKey:StatusID"`
	ServicePhaseID uuid.UUID   `json:"servicePhaseId" gorm:"type:uuid;not null"`
	ServicePhase  ServicePhase `json:"servicePhase" gorm:"foreignKey:ServicePhaseID"`
	CreatedAt     time.Time    `json:"createdAt" gorm:"autoCreateTime"`
	UpdatedAt     time.Time    `json:"updatedAt" gorm:"autoUpdateTime"`
} 